### PR TITLE
Add spider predator option

### PR DESCRIPTION
**Pull request by kswenson**
Add spider predator option [#157796779]
- add configuration panel for predator traits

### CODE REVIEW

File: src/species/spider.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  new Trait({ name: 'eating distance', default:  5 }),
+  new Trait({ name: 'mating distance', default:  5 }),
+  new Trait({ name: 'max offspring',   default:  2 }),
+  new Trait({ name: 'resource consumption rate', default:  10 }),
+  new Trait({ name: 'metabolism', default:  1 })
+];
+
+class Spider extends BasicAnimal {
+  step() {
+    console.log();
```
[the remark this answers is not in the document]
Comment (kswenson) [reply src/species/spider.ts]: Good catch! And that's why we have code reviews before merging to master. 😉

File: src/components/corn-chart.tsx [hunk 2]
```diff
@@ -26,12 +26,18 @@ export default class CornChart extends React.Component<IProps, IState> {
...
+    // always show five years of data
+    while (cornData.length < 5) {
+      cornData.push({ year: `Year ${cornData.length + 1}`, planted: 0, harvested: 0 });
+    }
+
     return (
-    	<BarChart width={300} height={180} data={cornData} barGap={0}
+      <BarChart className="corn-chart"
+            width={width} height={height} data={cornData} barGap={0}
             margin={{top: 5, right: 30, left: 20, bottom: 5}}>
```
Comment (dstrawberrygirl) [inline src/components/corn-chart.tsx]: Do these charts require in-line styling or can this be moved to css? Not hugely urgent for this phase of development, mostly curious about this library.

File: src/components/spider-trait-panel.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    };
+
+    return (
+      <div className="spider-trait-panel">
+        {renderConfigTrait(
+          "Default speed:",
+          "Controls the default speed of the spider",
+          "trait-default-speed")}
+        {renderConfigTrait(
+          "Speed:",
```
Comment (dstrawberrygirl) [inline src/components/spider-trait-panel.tsx]: I believe speed varies by behavior and age - perhaps take this parameter out and only expose default speed.

File: src/components/corn-chart.tsx (hunk 2 above)
Comment (kswenson) [reply src/components/corn-chart.tsx]: Good question. I didn't take the time to investigate. At that point in the process I was just looking for the quickest/easiest way to get a bar chart onto the screen. 😉

**Pull request merged**